lib/jormungandr/src/Cardano/Wallet/Jormungandr/Compatibility.hs

```diff
@@ -69,8 +74,18 @@ genesis = BlockHeader
 instance TxId (Jormungandr n) where
     txId = undefined
 
-instance KeyToAddress (Jormungandr n) where
-    keyToAddress = undefined
+instance forall n. KnownNetwork n => KeyToAddress (Jormungandr n) where
+    keyToAddress key = Address
```

Comment:
```suggestion
    keyToAddress = Address . addressToByteString . getKey
```

I would prefer this to be encapsulated in a function which lives in the Binary module. So that the Put stuff doesn't leak out. Something like:

```haskell
addressToByteString :: XPub -> ByteString
addressToByteString = BL.toStrict . runPut . putSingleAddress
```

Up to you though.

Reply:
Agreed with @rvl here :+1:

Reply:
I went with

```haskell
singleAddressFromKey :: forall n. KnownNetwork n => Proxy n -> XPub -> Address
singleAddressFromKey _ xpub = Address $ BL.toStrict $ runPut $ do
    putWord8 (single @n)
    putByteString (unXPub xpub)
```

--> 5bc2157

Considered naming it something like `newSingleAddress` too.